apps/s_cb.c

```diff
@@ -1304,29 +1351,37 @@ static int security_callback_debug(const SSL *s, const SSL_CTX *ctx,
     security_debug_ex *sdb = ex;
     int rv, show_bits = 1, cert_md = 0;
     const char *nm;
+    int show_nm;
     rv = sdb->old_cb(s, ctx, op, bits, nid, other, ex);
     if (rv == 1 && sdb->verbose < 2)
         return 1;
     BIO_puts(sdb->out, "Security callback: ");
 
     nm = lookup(op, callback_types, NULL);
+    show_nm = nm!=NULL;
     switch (op) {
     case SSL_SECOP_TICKET:
     case SSL_SECOP_COMPRESSION:
         show_bits = 0;
-        nm = NULL;
+        show_nm = 0;
         break;
     case SSL_SECOP_VERSION:
         BIO_printf(sdb->out, "Version=%s", lookup(nid, ssl_versions, "???"));
         show_bits = 0;
-        nm = NULL;
+        show_nm = 0;
         break;
     case SSL_SECOP_CA_MD:
     case SSL_SECOP_PEER_CA_MD:
         cert_md = 1;
         break;
+    case SSL_SECOP_SIGALG_SUPPORTED:
+    case SSL_SECOP_SIGALG_SHARED:
+    case SSL_SECOP_SIGALG_CHECK:
+    case SSL_SECOP_SIGALG_MASK:
+        show_nm = 0;
+        break;
     }
-    if (nm != NULL)
+    if (show_nm)
```

Comment:
Why is this change necessary?

Reply:
Hi, thank you for checking the code; in this case variable _'nm'_ does contain a literal I wish to print; problem is line 1385 (line 1330 in original) which prints an =equal sign after nm: 
```
-    if (nm != NULL)
+    if (show_nm)
         BIO_printf(sdb->out, "%s=", nm);
```
this doesn't fit into the new behaviour (it either prints 'scheme=xxx' or 'algorithm=xxx, hash=yyy'), so I had to find a way to suppress printing of 'nm' without destructing its value.

Another option would be calling `nm = lookup(op, callback_types, NULL);` once more in `case SSL_SECOP_OTHER_SIGALG`.

Reply:
Though 'nm' can still be NULL when I wish to print it, so I added one more check:
```
            if (nm != NULL) BIO_printf(sdb->out, "%s", nm);
            else            BIO_printf(sdb->out, "s_cb.c:security_callback_debug op=0x%x", op);
```

Reply:
Ok